components/apimgt/org.wso2.carbon.apimgt.rest.api.commons/src/main/java/org/wso2/carbon/apimgt/rest/api/common/interceptors/RESTAPISecurityInterceptor.java

```diff
@@ -118,6 +119,15 @@ public boolean preCall(Request request, Response response, ServiceMethodInfo ser
                 response.setStatus(javax.ws.rs.core.Response.Status.OK.getStatusCode()).setEntity(Json.pretty
                         (swagger)).setMediaType(MediaType.APPLICATION_JSON).send();
                 return false;
+            } else if (requestURI.contains("swagger.yaml")) {
+                try {
+                    yamlContent = RestApiUtil.getStoreRestAPIResource();
+                } catch (APIManagementException e) {
+                    log.error("Couldn't find swagger.yaml for store", e);
+                }
+                response.setStatus(javax.ws.rs.core.Response.Status.OK.getStatusCode()).setEntity(yamlContent)
```

Comment:
@nuwand Yes error responses are missing there I will add them, 
@malinthaprasan isn't it more appropriate to return 404, Since here an exception can be occurred due to an [IOError](https://github.com/wso2/carbon-apimgt/blob/master/components/apimgt/org.wso2.carbon.apimgt.rest.api.commons/src/main/java/org/wso2/carbon/apimgt/rest/api/common/util/RestApiUtil.java#L336) where the swagger file is not found in the file system. so that 404 would truly reflect the missing file(resource).
WDYT ?